courses/utils.py

```diff
@@ -0,0 +1,50 @@
+"""
+Utility functions for the courses app
+"""
+import re
+from opaque_keys.edx.keys import CourseKey
+from opaque_keys import InvalidKeyError
+
+NUMBER_SEASON_MAP = {
+    1: 'Spring',
+    2: 'Summer',
+    3: 'Fall'
+}
+
+
+def parse_edx_key(edx_course_key):
+    try:
+        course_key = CourseKey.from_string(edx_course_key)
+    except InvalidKeyError:
+        return None
+    return course_key.run if course_key else None
+
+
+def get_year_season_from_edx_key(parsed_course_key):
+    found = re.search('^[1-3]T[1|2][0-9]{3}$', parsed_course_key)
+    if found:
+        season_year_tuple = found.group().split('T')
+        return int(season_year_tuple[1]), int(season_year_tuple[0])
+    return None, None
+
+
+def get_year_season_from_dates(course_run):
+    if not course_run.start_date:
+        return None, None
+    start_month = course_run.start_date.month
+    if 1 <= start_month <= 4:
+        season_num = 1
```

Comment:
yeah, this is 'legacy'. i had an indexing structure that i thought would work for sorting these results directly in elasticsearch, but that didn't end up working out with searchkit. i can change these functions to return the season directly instead of the integer